cmd/tfimport/main.go

```diff
@@ -70,6 +73,25 @@ func run() error {
 		importRn = &runner.Default{}
 	}
 
+	retry := true
+	for retry {
+		retry, err = planAndImport(rn, importRn)
+		if err != nil {
+			return err
+		}
+
+		if retry {
+			log.Println("Succeeded in importing some resources, but others failed. Retrying the import, in case dependent values have now been populated.")
+		}
+	}
+
+	return nil
+}
+
+// This function does the full plan and import cycle.
+// If it imported some resources but failed to import others, it will return true for retry. This is a simple way to solve dependencies without having to figure out the graph.
```

Comment:
It only repeats when the function returns "true" for "retry", which only happens when we've imported something (made some progress) but failed to import some other things (aren't done yet). If it's just not possible to import everything (in any order), it will stop looping when it fails to import anything.

Reply:
Is `retry` only `true` when some resources is still in progress or some imports actually error out due to a legit reason?

Reply:
It's true only when some resources from the plan were imported successfully, and some were importable but failed to import.